src/pages/user-settings.tsx

```diff
@@ -248,49 +273,96 @@ const Settings: FunctionComponent<{
-                <Input name="privkey" placeholder="Paste token here" />
+                <Label htmlFor="description">Description</Label>
+                <Input {...register("description")} />
+
+                <div className="mt-4">
+                  <span>Select scopes</span>
+                  <span className="block text-sm text-gray-500">
+                    Scopes set the permissions of the token. You can choose
+                    multiple scopes.
+                  </span>
+                </div>
+
+                <div className="mt-4 flex items-center justify-between gap-2">
+                  <Label htmlFor="scan" className="flex-1">
+                    scan
+                    <span className="block text-sm text-gray-500">
+                      Use this token to scan your repositories.
+                    </span>
+                  </Label>
+                  <Switch
+                    onCheckedChange={(e) => setValue("scan", e)}
+                    checked={Boolean(watch("scan"))}
+                  />
+                </div>
+
+                <div className="mt-4 flex items-center justify-between gap-2">
+                  <Label htmlFor="manage" className="flex-1">
+                    manage
+                    <span className="block text-sm text-gray-500">
+                      Use this token to manage your repositories.
+                    </span>
+                  </Label>
+                  <Switch
+                    onCheckedChange={(e) => setValue("manage", e)}
+                    checked={Boolean(watch("manage"))}
+                  />
+                </div>
               </CardContent>
               <CardFooter className="flex justify-end">
-                <Button type="submit" variant="destructiveOutline">
-                  Revoke
-                </Button>
+                <div className="mt-6 flex flex-row justify-end">
+                  <Button type="submit">Create</Button>
+                </div>
               </CardFooter>
             </form>
           </Card>
+          {personalAccessTokens.map((pat) =>
+            "privKey" in pat ? (
+              <React.Fragment key={pat.fingerprint}>
+                <Card>
+                  <CardHeader>
+                    <CardTitle>{pat.description} </CardTitle>
+                  </CardHeader>
+                  <CardContent className="pt-6">
+                    <CopyCode
+                      codeString={pat.privKey}
+                      language="shell"
+                    ></CopyCode>
+                    <span className="mt-2 block text-sm text-destructive">
+                      Make sure to copy the token. You won&apos;t be able to see
+                      it ever again
+                    </span>
+                  </CardContent>
+                </Card>
+              </React.Fragment>
+            ) : null,
+          )}
+
           <div className="mb-6 flex flex-col gap-4">
             {personalAccessTokens.map((pat) =>
```

Comment:
i think this can be improved by filtering beforehand: 


```suggestion
            {personalAccessTokens.filter(p => "privKey" in pat).map((pat) =>
```